src/Processors/Merges/Algorithms/RowRef.h

```diff
@@ -122,6 +128,29 @@ inline void intrusive_ptr_release(SharedChunk * ptr)
         ptr->allocator->release(ptr);
 }
 
+inline boost::intrusive_ptr<detail::SharedChunk> SharedChunk::cloneForSelectedRow(size_t row_num)
```

Comment:
how often we could execute this? like once per block_size rows?

Reply:
It depends on the data, but in general case yes, once per block (to record the last row of that block).

Reply:
I Implemented a different approach and get rid of this.